### ISSUE TITLE
not able to install GeoIP2::Database::Reader

### ISSUE THREAD

**Issue by thvv** [maxmind/GeoIP2-perl#60]
Since GeoIP Legacy will not be supported after April, I wanted to
update my software to GeoIP2.  My environment is macOS 10.12.6, Perl 5.24

I did "sudo -H cpan" and then "install GeoIP2::Database::Reader" and install failed.

MAXMIND/GeoIP2-004000 required Data::Validate::IP
DROLSKY/Data-Validate-IP.0.27 required NetAddr::IP
MIKER/NetAddr-IP.4.079 failed its tests and would not install.

      Test Summary Report
      -------------------
      t/binet_ntoa.t            (Wstat: 0 Tests: 20 Failed: 3)
	Failed tests:  7, 9, 11
      t/binet_pton.t            (Wstat: 0 Tests: 32 Failed: 3)
	Failed tests:  19, 21, 23
      t/inet_pton.t             (Wstat: 0 Tests: 32 Failed: 3)
	Failed tests:  19, 21, 23
      t/ipv6_ntoa.t             (Wstat: 0 Tests: 20 Failed: 3)
	Failed tests:  7, 9, 11
      Files=39, Tests=1528,  2 wallclock secs ( 0.21 usr  0.11 sys +  0.87 cusr  0.23 csys =  1.42 CPU)
      Result: FAIL
      Failed 4/39 test programs. 12/1528 subtests failed.
      make[2]: *** [test_dynamic] Error 255
      make[1]: *** [subdirs-test_dynamic] Error 2
      make: *** [subdirs-test_dynamic] Error 2
	MIKER/NetAddr-IP-4.079.tar.gz
	/usr/bin/make test -- NOT OK
      //hint// to see the cpan-testers results for installing this module, try:
	reports MIKER/NetAddr-IP-4.079.tar.gz
	DROLSKY/Data-Validate-IP-0.27.tar.gz
	Has already been unwrapped into directory /var/root/.cpan/build/Data-Validate-IP-0.27-0
	DROLSKY/Data-Validate-IP-0.27.tar.gz
	Has already been prepared
      Running make for D/DR/DROLSKY/Data-Validate-IP-0.27.tar.gz
      Warning: Prerequisite 'NetAddr::IP => 4' for 'DROLSKY/Data-Validate-IP-0.27.tar.gz' failed when processing 'MIKER/NetAddr-IP-4.079.tar.gz' with 'make_test => NO'. Continuing, but chances to succeed are limited.

(i tried the "reports" command and it failed with an exception.)

**Comment by oschwald** [comment]
I think this is the same as [this `NetAddr::IP` issue](https://rt.cpan.org/Public/Bug/Display.html?id=88635). Perhaps you could try upgrading `Socket6`?

**Comment by thvv** [comment]
> On Jan 17, 2018, at 4:12 PM, Gregory Oschwald <notifications@github.com> wrote:
> I think this is the same as this NetAddr::IP issue. Perhaps you could try upgrading Socket6?

thanks for the valuable clue.  i need another though.

the last line of the NetAddr::IP issue says "Fixed in 0.26 added"
so in cpan i did "i Socket6" .. mine says INST_VERSION 0.28 so I guess I have this fix.

i ran cpan-outdated.  it showed Socket as downrev, updated that.
NetAddr::IP still does not install.

what else can i try?

**Comment by thvv** [comment]
I solved my problem.  My CPAN config was looking at a mirror that was over 400 days out of date.
Changed the mirror, reloaded the index, tried again, and the install completed after a lot of updates.

**Comment by oschwald** [comment]
Thanks for letting us know. I will close this then.

**Comment by thvv** [comment]
> On Jan 17, 2018, at 4:12 PM, Gregory Oschwald <notifications@github.com> wrote:
> 
> I think this is the same as this NetAddr::IP issue. Perhaps you could try upgrading Socket6?
> 

I resolved this problem by upgrading Perl.

I am on a Mac and did a "port install ImageMagick" which 
upgraded a whole lot of software, including Perl.
It upgraded Perl 5.24.3 to 5.24.4, and also installed 5.24.6.

Then I did a "sudo -H cpan" and "install GeoIP2::Database::Reader"
This took about 45 minutes and installed a whole kitchen sink of stuff.
Started with NetAddr::IP, which succeeded this time, and then went on and on and on.
Finally it finished without failing.  Looks like it installed in the 5.26 library.
Guess I have to change PERL5LIB to make use of all this.
